src/tokens.js

```diff
@@ -271,101 +272,288 @@ const tokens = {
+    return transaction.sendTransaction(newTxData, pin);
+  },
+
+  /**
+   * Get inputs from the amount to be melted
+   *
+   * @param {number} amount Amount of the token to get the inputs
+   * @param {string} token Token uid that will be melted
+   *
+   * @return {Object} Object with {'inputsAmount': the total amount in the returned inputs, 'inputs': Array of inputs ({'tx_id', 'index', 'address', 'token'})} or null, if does not have this amount
+   *
+   * @memberof Tokens
+   * @inner
+   */
+  getMeltInputs(amount, token) {
+    const data = wallet.getWalletData();
+    // If wallet has no data yet, return null
+    if (data === null) {
+      return null;
+    }
+
+    const inputs = [];
+    let inputsAmount = 0;
+    // Get history for this token
+    const filteredHistory = wallet.filterHistoryTransactions(data.historyTransactions, token, false);
+
+    for (const tx of filteredHistory) {
+      if (tx.is_voided) {
+        // Ignore voided transactions.
+        continue;
+      }
+      for (const [index, txout] of tx.outputs.entries()) {
+        if (wallet.isAuthorityOutput(txout)) {
+          // Ignore authority outputs.
+          continue;
+        }
+        // If output is still not spent, and is from this token, and is mine, add to inputs array and sum the value
+        if (txout.spent_by === null && txout.token === token && wallet.isAddressMine(txout.decoded.address, data)) {
+          inputs.push({'tx_id': tx.tx_id, 'index': index, 'token': token, 'address': txout.decoded.address});
+          inputsAmount += txout.value;
+
+          if (inputsAmount >= amount) {
+            // If reached the requested amount, return
+            return {inputs, inputsAmount};
           }
-        }, (e) => {
-          // Error in request
-          reject(e.message);
-        });
-      } catch (e) {
-        if (e instanceof AddressError || e instanceof OutputValueError) {
-          reject(e.message);
-        } else {
-          // Unhandled error
-          throw e;
         }
       }
-    });
-    return promise;
+    }
+
+    return null;
+  },
+
+  /**
+   * Create delegate authority data
+   *
+   * @param {string} txID Hash of the transaction to be spent
+   * @param {number} index Index of the output being spent
+   * @param {string} addressSpent Address of the output being spent
+   * @param {string} token Token uid to be delegated the authority
+   * @param {string} address Destination address of the delegated authority output
+   * @param {boolean} createAnother If should create another authority output for this wallet, after delegating this one
+   * @param {string} type Authority type to be delegated ('mint' or 'melt')
+   *
+   * @return {Object} Delegate authority data {'inputs', 'outputs', 'tokens'}
+   *
+   * @memberof Tokens
+   * @inner
+   */
+  getDelegateAuthorityData(txID, index, addressSpent, token, address, createAnother, type) {
```

Comment:
I would rename it to `createDelegateAuthorityData` (instead of get).

Reply:
Done